### ISSUE TITLE
Allow using single file with root CA certificates

### ISSUE THREAD

**Issue by Sokolmish** [astibal/smithproxy#33]
Hello. I get the following error when i try to make an https connection via Smithproxy.
```
Detected application: cat='www', name='http2/start'
original TLS peer verification failedTLS content replacement
serverside connection closed: 0::ssli_173.194.221.101:443
0::ssli_173.194.221.101:443   dropped by proxy:(ssl:Server certificate is issued by untrusted certificate authority)(ssl: enforced)
Connection stop
```

This problem can be caused by the fact, that root certificates on my system are stored in the single file, instead of multiple files in `/etc/ssl/certs`.
In the config file there is an option to specify path to certificates directory, but not to the file. Also, [there](https://github.com/astibal/socle/blob/cabc5e4ea9ead30b15c1ad264074f77c4e9297e5/sslcertstore.cpp#L379) `SSL_CTX_load_verify_locations()` is used only with `CApath` and without `CAfile`.

I think that there can be 2 options in the config file: for file and directory paths (or both of them simultaneously), so one can use Smithproxy on more systems.

**Comment by Sokolmish** [comment]
Also, I found mentioned error message in `*.pcapng` file, but there wasn't any error message in the log and I got very strange behaviour  on the client. Is it ok?

```
root@a0d913ebe874:~# curl --cacert ./smith_ca.cert https://www.google.com
curl: (16) Error in the HTTP2 framing layer
# Verbose error is: http2 error: Remote peer returned unexpected data while we expected SETTINGS frame.  Perhaps, peer does not support HTTP/2 properly.

root@a0d913ebe874:~# curl --cacert ./smith_ca.cert https://info.cern.ch
<html><head><meta http-equiv="Refresh" content="0; url=/SM/IT/HP/RO/XY/warning?q=1"></head><body></body></html>
```

**Comment by astibal** [comment]
Hi, 
thanks for report, interestingly enough, I wanted to add single-CA file support, too. :) This is good idea of course.

The latter is basically a missing feature - HTTP 1.1 replacement for information the site has some TLS problem (ie. self-signed cert, invalid, or other issue). By default, smithproxy returns HTTP(s) replacement message about that fact. 
That being said, it's a TLS issue **not** between you and smithproxy, but between smithproxy and the server.

To summarize, the second problem is caused by the first issue you have reported. 

I will be looking into both of them quite soon.

**Comment by astibal** [comment]
I created #34 to cover HTTP/2 incorrect replacement with HTTP 1 response. Let's keep this to track single-file CA bundle NFR.

**Comment by astibal** [comment]
Hi, single-file CA bundle is now supported (since `0.9.31`). CA bundle file is preferred, if set. 
You  can specify the file in `settings/ca_bundle_file` variable.

Sadly, I didn't include it in Release Notes -  I will update the document soon.
